src/App.tsx

```diff
@@ -1,26 +1,124 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { useEffect, useRef, useState } from 'react';
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer/Footer';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './types/Todo';
+import { deleteTodo, getTodos, USER_ID } from './api/todos';
+import { Errors } from './utils/Errors';
+import { FilterTodosBy } from './utils/FilterTodosBy';
+import { Error } from './components/Error/Error';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [hasError, setHasError] = useState<Errors>(Errors.NoError);
+  const [filterBy, setFilterBy] = useState<FilterTodosBy>(FilterTodosBy.All);
+  const [title, setTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isDeleting, setIsDeleting] = useState<boolean>(false);
+  const [isDeleteAllCompleted, setIsDeleteAllCompleted] =
+    useState<boolean>(false);
```

Comment:
it's redundant for boolean values

```suggestion
  const [isDeleting, setIsDeleting] = useState(false);
  const [isDeleteAllCompleted, setIsDeleteAllCompleted] =
    useState(false);
```